### PR TITLE
optimize `for` loop in WildcardPermission

### PR DESCRIPTION
**Pull request by SubhanSh**

### CODE REVIEW

**Comment by systemsolutionweb** [comment]
Why change `static` to self??

**Comment by angeljqv** [comment]
> Why change `static` to `self`??

Yes, he is introducing a issue

**Comment by angeljqv** [comment]
> Why change `static` to `self`??

Yes, he is introducing a issue

**Comment by angeljqv** [comment]
> Why change `static` to `self`??

Yes, he is introducing a issue

**Comment by angeljqv** [comment]
Hi??

**Comment by angeljqv** [comment]
Hi?

File: src/WildcardPermission.php [hunk 1]
```diff
@@ -60,8 +61,8 @@ public function implies($permission): bool
             $i++;
         }
 
-        for ($i; $i < $this->parts->count(); $i++) {
-            if (! $this->parts->get($i)->contains(static::WILDCARD_TOKEN)) {
+        for ($i; $i < $partsCount; $i++) {
+            if (! $this->parts->get($i)->contains(self::WILDCARD_TOKEN)) {
```
Comment (drbyte) [inline src/WildcardPermission.php:65]: Why did you change `static` to `self`?
